### ISSUE TITLE
Radarr can not update anymore under DSM 7.1

### ISSUE THREAD

**Issue by synofan** [SynoCommunity/spksrc#5219]
### Is this a new Bug?

- [X] I checkd that the bug hasn't been reported before

### Package Name

Radarr

### Package Version

4.0.5.5981

### Device Model

DS1621+ and virtual DSM 7.1

### Device Architecture

x86_64

### Firmware Version

7.1

### What happened?

[Tracelog Radarr DSM 7_1.txt](https://github.com/SynoCommunity/spksrc/files/8530601/Tracelog.Radarr.DSM.7_1.txt)

Access to tmp directory is denied, update fails

### Reproduction steps

```bash
1. under virtual dsm 7.0 update succesfull
2. under virtual dsm 7.1 update fails
3. this is the same for native DSM 7.1
...
```

### Install Log

```log
not available
```

**Comment by synofan** [comment]
additional info : update log is not created

**Comment by john5du** [comment]
Same issues.

**Comment by kunude** [comment]
you can down this spk files @ https://github.com/SynoCommunity/spksrc/suites/6174998010/artifacts/217575879

**Comment by synofan** [comment]
Works like a charm THX

**Comment by bakerboy448** [comment]
The root cause appears to be that /tmp is no longer executable on DSM7.1

the alt tmp dir flag in the package is required, but it appears due to other unrelated breaking changes in the commit that added that flag that package (Oct 2021) is not available for most? All? Users

**Comment by BenjV** [comment]
What do you mean by no longer executable, /tmp is a folder and is and was never executable.
And if you ment accessible instead of executable, that is also not the case.
On DSM 7.1 the /tmp folder still has 1777 permissions zetting so accessible for everybody.

**Comment by bakerboy448** [comment]
It's been confirmed by the Radarr Team that synology DSM 7.1 does not allow executing apps/files from /tmp anymore.

The Updater fails to run.

**Comment by ta264** [comment]
> On DSM 7.1 the /tmp folder still has 1777 permissions zetting so accessible for everybody.

Yes, this is true, but DSM won't allow Radarr to execute files downloaded to `/tmp`.  A custom build with an alternative `TMPDIR` functions correctly.  I don't know how DSM is doing this, but it is.

I am working on forking `spksrc` to produce working *arr packages for v4 and DSM 7.1 - I think it's probably easier to keep it under our control so that we can make the fixes required by .NET upgrades without having to bother you guys every time.

Once I've got something that's working I can work with you guys to upstream my fixes if you're interested.

**Comment by hgy59** [comment]
@ta264 I can easyly update the radarr package with use of alternate tmp dir (it uses SYNOPKG_PKGTMP i.e. /var/packages/radarr/tmp), but I am not aware whether the package update must prevent to downgrade radarr binaries that might be updated by internal updater (similar to #5181 for syncthing).
Another open issue in  #5211 states that all user data is lost on update.

**Comment by hgy59** [comment]
Created #5252 to finally publish radarr with alternate TMPDIR.

To test:
- [ ] validate package update after radarr internal update
- [ ] validate content user data (var folder) is not lost on update

**Comment by ta264** [comment]
Thanks @hgy59 

There were some other changes made to the radarr spk around the same time as the TMPDIR fix which broke user data.

In any case it looks like the glibc version on DSM 6 and some DSM 7 archs is too old to support .NET 6 natively.  I am working on a chroot-style solution to let us run Radarr inside a debian 10 rootfs.  Initial results are promising.  

If you're up for assisting it would be great to have your input - most of the development chat is on our discord.

**Comment by hgy59** [comment]
> In any case it looks like the glibc version on DSM 6 and some DSM 7 archs is too old to support .NET 6 natively. I am working on a chroot-style solution to let us run Radarr inside a debian 10 rootfs. Initial results are promising.

We have already published a .NET 6 runtime package. For DSM 6 the package contains a newer stdlibc++ and patches the dotnet executable to use the library in the package. For DSM 7 (except probably comcerto2k archs) the stdlibc++ of the system is usable by .NET 6.

**Comment by synofan** [comment]
See a new version of Radarr in package Centre, made a backup (lucky me) applied the update from Package Centre, it starts, but no movies with message displayed about faulty language (dont have exact message anymore) decided to remove package (fully) reinstalled looked at system info...ha there is an update, executed the update. This time flawless with a generated update log. restored the backup and voila, works like a charm. THX for all the hard work :-). Big thumbs up.

**Comment by synofan** [comment]
by the way version seen is v20220510-16 at this moment.

**Comment by ta264** [comment]
@hgy59 there's some sort of bug in the synocommunity Radarr package where the switch from the old to new style `PKGVAR` isn't handled properly.  I think this is part of the reason the update got pulled last time.

**Comment by synofan** [comment]
FYI Have to say i had the temporary fix package running  https://github.com/SynoCommunity/spksrc/suites/6174998010/artifacts/217575879 so dont know if that was making the problems. Anyhow with a backup and a clean reinstall it works great.

**Comment by hgy59** [comment]
finally fixed with #5268
